perpVects/QFPHelpers.cpp

```diff
@@ -37,19 +37,51 @@ std::ostream& operator<<(std::ostream& os, const unsigned __int128 &i){
   return os;
 }
 
-std::string
-getSortName(sort_t val){
-  switch(val){
-  case lt:
-    return "lt";
-  case gt:
-    return "gt";
-  case bi:
-    return "bi";
-  case def:
-    return "us";
-  default:
-    return "something bad happened, undefined sort type";
+namespace FPHelpers{
+    auto
+  swap_float_int(unsigned int val) -> float{
```

Comment:
Good call on uint type

On Wednesday, August 24, 2016, Michael Bentley <notifications@github.com>
wrote:

> In perpVects/QFPHelpers.cpp
> <https://github.com/Geof23/QFP/pull/13#discussion_r76152854>:
>
> > -std::string
> > -getSortName(sort_t val){
> > -  switch(val){
> > -  case lt:
> > -    return "lt";
> > -  case gt:
> > -    return "gt";
> > -  case bi:
> > -    return "bi";
> > -  case def:
> > -    return "us";
> > -  default:
> > -    return "something bad happened, undefined sort type";
> > +namespace FPHelpers{
> > +    auto
> > +  swap_float_int(unsigned int val) -> float{
>
> Why not use uint32_t here? Unsigned int is not a specific size according
> to IEEE or the C++ standard. For integral types, you should really use the
> specific sizes you intend as to not cause differences from architectures
> (such as if we go to a 32-bit architecture).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Geof23/QFP/pull/13/files/57db3da0327a58ab91d6a9ee1c70501608c4db5b#r76152854>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFiXD2naCNBtV85zo2wbOOw7eLYTBIbIks5qjMgmgaJpZM4Jr1FF>
> .
>